src/array_api_typing/_array.py

```diff
@@ -67,10 +68,131 @@ def dtype(self, /) -> DTypeT_co:
         ...
 
 
+class HasDevice(Protocol):
```

Comment:
Should these be public or just Array?

Reply:
`HasDType` could be considered special in that we want it to be a type parameter.

Reply:
> Should these be public or just Array?

It's as straightforward as can be, so I don't see any risk in making it public

Reply:
Although, without a generic type parameter, it probably won't be very useful in practice. But I suppose we could also add that later if we need it.